llvm/lib/CodeGen/GlobalISel/CombinerHelper.cpp

```diff
@@ -316,6 +316,54 @@ adderGenerator(const int32_t From, const int32_t To, const int32_t StepSize) {
   };
 }
 
+Register CombinerHelper::createUnmergeValue(
+    MachineInstr &MI, const Register SrcReg, const Register DstReg,
+    const uint8_t DestinationIndex, const uint32_t Start, const uint32_t End) {
+  Builder.setInsertPt(*MI.getParent(), MI);
+  const LLT DstTy = MRI.getType(DstReg);
+  const LLT SrcTy = MRI.getType(SrcReg);
+  assert((!DstTy.isVector() ||
```

Comment:
DeMorgan doesn't do much, since `(!x || y) && z` doesn't really turn into something nice if you negate it. Instead I think it is better if I check for scalars directly instead of checking if it is not a vector.